src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -2014,7 +2014,20 @@ fn disambiguator_error(
     link_range: Range<usize>,
     msg: &str,
 ) {
-    report_diagnostic(cx.tcx, BROKEN_INTRA_DOC_LINKS, msg, item, dox, &link_range, |_diag, _sp| {});
+    report_diagnostic(cx.tcx, BROKEN_INTRA_DOC_LINKS, msg, item, dox, &link_range, |diag, _sp| {
+        let channel = match env!("CFG_RELEASE_CHANNEL") {
+            "stable" => env!("CFG_RELEASE_NUM"),
+            "beta" => "beta",
+            "nightly" | "dev" => "nightly",
+            // custom build of rustdoc maybe? link to the stable docs just in case
+            _ => "",
+        };
```

Comment:
nit: Can you extract a function for this? I could imagine this code being useful elsewhere, and it's not specific to reporting a disambiguator error. Maybe put it in `src/librustdoc/core.rs`?

Reply:
Excellent catch, it turns out a lot of rustdoc was either rewriting this or ignoring it and hard-coding nightly/std. I added a function and used it in a bunch of places :)